src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel.cs

```diff
@@ -0,0 +1,436 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
this is the meat of the change.  it follows the design of NavBars.  Basically, we split work into computation/updatng of the UI, and then figuring out what to select based on character position.